db/view/view.cc

```diff
@@ -1126,6 +1131,9 @@ void view_updates::generate_update(
         const clustering_or_static_row& update,
         const std::optional<clustering_or_static_row>& existing,
         gc_clock::time_point now) {
+    if (_update_done) {
+        return;
+    }
```

Comment:
Perhaps we could turn this into `if (_view_info.has_same_partition_key()) { return; }`? We wouldn't have to depend on the new variable.

Reply:
It's not a sufficient condition to skip the update
It needs to be also the case that we have generated a partition tombstone already and we're generating now row tombstones
For example, this function can be called also when we're doing insertion
But I will try to think if I can make this more clear

Reply:
Hmm, right, I see.

I can see that, later, we have this part:

```c++
// The view key is necessarily the same pre and post update.
if (existing && existing->is_live(*_base)) {
    if (update.is_live(*_base)) {
        update_entry(db, base_key, update, *existing, now);
    } else {
        delete_old_entry(db, base_key, *existing, update, now);
    }
}
```

Perhaps we could make `delete_old_entry` do the check?

Reply:
maybe in `delete_old_entry` we can check if we're writing a row tombstone and we already have an update for the partition tombstone which is equal or better, then we can skip this
I feel this is a little bit more delicate, I will try to look

Reply:
I think I managed to remove the `_update_done` variable.
Now in `get_view_rows` I skip creating the row if there is a partition tombstone update, so there won't be updates generated for the row.